### PR TITLE
Ncert tb search

### PR DESCRIPTION
**Pull request by PradyumnaNagendra**

### CODE REVIEW

File: ansible/roles/search-service-deploy/templates/search-service.conf.j2 [hunk 1]
```diff
@@ -106,4 +106,5 @@ language.map={"Hindi":"hi", "English":"en", "Telugu":"te", "Kannada":"ka", "Tami
 channel.default="in.ekstep"
 content.tagging.backward_enable=true
 content.tagging.property=["subject","medium"]
-compositesearch.index.name="{{ compositesearch_index_name }}"
\ No newline at end of file
+compositesearch.index.name="{{ compositesearch_index_name }}"
+relatedBoards.props=["board"]
```
Comment (maheshkumargangula) [inline ansible/roles/search-service-deploy/templates/search-service.conf.j2]:
```suggestion
content.relatedBoards.props=["board"]
```

File: searchIndex-platform/module/search-actors/src/main/java/org/ekstep/search/actor/SearchManager.java [hunk 2]
```diff
@@ -266,6 +267,8 @@ private SearchDTO getSearchDTO(Request request) throws Exception {
 			searchObj.setSortBy(sortBy);
 			searchObj.setFacets(facets);
 			searchObj.setProperties(properties);
+			// Added Implicit Filter Properties To Support NCERT Textbook
```
Comment (maheshkumargangula) [inline searchIndex-platform/module/search-actors/src/main/java/org/ekstep/search/actor/SearchManager.java]:
```suggestion
			// Added Implicit Filter Properties To Support Collection content tagging to reuse by tenants.
```

File: searchIndex-platform/module/search-actors/src/main/java/org/ekstep/search/actor/SearchManager.java [hunk 3]
```diff
@@ -743,4 +745,17 @@ private void getAggregations(Map<String, Object> req, SearchDTO searchObj) {
 
 	}
 
+	private void setImplicitFilters(String objectType, Map<String, Object> filters, SearchDTO searchObj) throws Exception {
+		List<String> consumerProps = Platform.config.hasPath("relatedBoards.props") ?
+				Platform.config.getStringList("relatedBoards.props") : new ArrayList<String>();
+		Map<String, Object> implicitFilter = new HashMap<String, Object>();
+		if (CollectionUtils.isNotEmpty(consumerProps) && MapUtils.isNotEmpty(filters) &&
```
Comment (maheshkumargangula) [inline searchIndex-platform/module/search-actors/src/main/java/org/ekstep/search/actor/SearchManager.java]: remove the condition on objectType.

File: searchIndex-platform/module/search-actors/src/test/java/org/ekstep/search/test/BaseSearchActorsTest.java [hunk 4]
```diff
@@ -159,7 +171,7 @@ private static void addToIndex(String uniqueId, Map<String, Object> doc) throws
 				CompositeSearchConstants.COMPOSITE_SEARCH_INDEX_TYPE, uniqueId, jsonIndexDocument);
 	}
 	
-	private static Map<String, Object> getContentTestRecord(String id, int index) {
+	private static Map<String, Object> getContentTestRecord(String id, int index, String channel) {
```
Comment (maheshkumargangula) [inline searchIndex-platform/module/search-actors/src/test/java/org/ekstep/search/test/BaseSearchActorsTest.java]: change the parameter to use `board`

File: searchIndex-platform/module/search-actors/src/test/resources/application.conf [hunk 5]
```diff
@@ -27,3 +27,5 @@ SearchActorSystem {
 		}
 	}
 }
+
+relatedBoards.props=["channel","board","subject","medium"]
```
Comment (maheshkumargangula) [inline searchIndex-platform/module/search-actors/src/test/resources/application.conf]:
```suggestion
content.relatedBoards.properties=["channel","board","subject","medium"]
```

File: ansible/roles/search-service-deploy/templates/search-service.conf.j2 (hunk 1 above)
Comment (PradyumnaNagendra) [reply ansible/roles/search-service-deploy/templates/search-service.conf.j2]: done

File: searchIndex-platform/module/search-actors/src/main/java/org/ekstep/search/actor/SearchManager.java (hunk 2 above)
Comment (PradyumnaNagendra) [reply searchIndex-platform/module/search-actors/src/main/java/org/ekstep/search/actor/SearchManager.java]: change is commited

File: searchIndex-platform/module/search-actors/src/main/java/org/ekstep/search/actor/SearchManager.java (hunk 3 above)
Comment (PradyumnaNagendra) [reply searchIndex-platform/module/search-actors/src/main/java/org/ekstep/search/actor/SearchManager.java]: Removed

**Pull request merged**